packages/jest-environment-node/src/index.ts

```diff
@@ -110,6 +116,20 @@ class NodeEnvironment implements JestEnvironment {
     }
     return null;
   }
+
+  compileFunction(code: string, params: Array<string>, filename: string) {
+    if (this.context) {
+      return compileFunction(code, params, {
+        filename,
+        parsingContext: this.context,
+      }) as any;
+    }
+    return null;
+  }
+}
+
+if (typeof compileFunction !== 'function') {
+  delete NodeEnvironment.prototype.compileFunction;
```

Comment:
could potentially do some `supportsCompileFunction(): boolean` instead, but I think this is fine

Reply:
Wouldn't it be more straight forward to make it a noop instead of deleting? Or there's some other advantage I don't see?

Reply:
we need to use `runScript` if `compileFunction` is missing, how would we fall back to that? differentiating between `null` and `undefined` return seems flaky

Reply:
Just checked back on the code in `jest-runtime` and you're right, this seems ok. Can you leave a comment on why we delete that and add a todo to remove it when we drop Node 8?